src/app/components/Modals/UnbondWithdrawModal.tsx

```diff
@@ -15,26 +16,37 @@ export const MODE_WITHDRAW = "withdraw";
 export type MODE = typeof MODE_UNBOND | typeof MODE_WITHDRAW;
 
 interface PreviewModalProps {
-  delegationHeight: number;
   open: boolean;
   onClose: (value: boolean) => void;
   onProceed: () => void;
   mode: MODE;
   awaitingWalletResponse: boolean;
+  delegationsAPI: DelegationInterface[];
+  txID: string;
 }
 
 export const UnbondWithdrawModal: React.FC<PreviewModalProps> = ({
-  delegationHeight,
   open,
   onClose,
   onProceed,
   mode,
   awaitingWalletResponse,
+  delegationsAPI,
+  txID,
 }) => {
   const { coinName, networkName } = getNetworkConfig();
   const versionInfo = useVersionInfo();
 
-  const globalParams = versionInfo?.currentVersion;
+  const delegation = delegationsAPI.find(
+    (delegation) => delegation.stakingTxHashHex === txID,
+  );
+  if (!delegation) {
+    throw new Error("Delegation not found");
+  }
```

Comment:
It's not good idea to throw errors inside render method. Do we already have error boundaries?

Reply:
yep you're right we don't have error boundaries yet. I have removed this. for now the display will be '-' if delegation isnt found.